auth/src/android/credential_android.cc

```diff
@@ -441,6 +449,50 @@ Credential EmailAuthProvider::GetCredential(const char* email,
   return cred;
 }
 
+//
+// PhoneAuthCredential methods
+//
+PhoneAuthCredential::PhoneAuthCredential() {}
+
+PhoneAuthCredential::PhoneAuthCredential(void* impl) {
+  impl_ = impl;
+  error_code_ = kAuthErrorNone;
+}
+
+PhoneAuthCredential::PhoneAuthCredential(const PhoneAuthCredential& rhs) {
+  *this = rhs;
+}
+
+PhoneAuthCredential& PhoneAuthCredential::operator=(
+    const PhoneAuthCredential& rhs) {
+  if (rhs.impl_ != nullptr) {
```

Comment:
I think you should remove the global reference to its own `impl_` first, if not nullptr, before assigning it to any other things, even assigning it to null.

To reduce unnecessary steps to add and remove references, you probably can check if two `PhoneAuthCredential` are the same before doing the rest of the work,.

Reply:
Huh....`Credential` is doing this as well. This looks......pretty wrong to me
https://github.com/firebase/firebase-cpp-sdk/blob/main/auth/src/android/credential_android.cc#L365

Reply:
I think this whole block can be reduced to 
```
PhoneAuthCredential& PhoneAuthCredential::operator=(
    const PhoneAuthCredential& rhs) {
  Credential::operator=(rhs);
  sms_code_ = rhs.sms_code_;
}
```

Also, please change `Credential::operator=()` to dereference before assigning `impl_`

Reply:
Great catch, thanks!